py/core/main/api/v3/collections_router.py

```diff
@@ -0,0 +1,782 @@
+                                 -H "Authorization: Bearer YOUR_API_KEY"
+                        """
+                        ),
+                    },
+                ]
+            },
+        )
+        @self.base_endpoint
+        async def add_document_to_collection(
+            id: UUID = Path(...),
+            document_id: UUID = Path(...),
+            auth_user=Depends(self.providers.auth.auth_wrapper),
+        ) -> WrappedMessageResponse:
+            """
+            Add a document to a collection.
+            """
+            if (
+                not auth_user.is_superuser
+                and id not in auth_user.collection_ids
+            ):
+                raise R2RException(
+                    "The currently authenticated user does not have access to the specified collection.",
+                    403,
+                )
+
+            result = await self.services[
+                "management"
+            ].assign_document_to_collection(document_id, id)
+            return result  # type: ignore
+
+        @self.router.get(
+            "/collections/{id}/documents",
+            summary="List documents in collection",
+            openapi_extra={
+                "x-codeSamples": [
+                    {
+                        "lang": "Python",
+                        "source": textwrap.dedent(
+                            """
+                            from r2r import R2RClient
+
+                            client = R2RClient("http://localhost:7272")
+                            # when using auth, do client.login(...)
+
+                            result = client.collections.list_documents(
+                                "123e4567-e89b-12d3-a456-426614174000",
+                                offset=0,
+                                limit=10,
+                                sort_by="created_at",
+                                sort_order="desc"
+                            )
+                        """
+                        ),
+                    },
+                    {
+                        "lang": "cURL",
+                        "source": textwrap.dedent(
+                            """
+                            curl -X GET "https://api.example.com/v3/collections/123e4567-e89b-12d3-a456-426614174000/documents?offset=0&limit=10&sort_by=created_at&sort_order=desc" \\
+                                 -H "Authorization: Bearer YOUR_API_KEY"
+                        """
+                        ),
+                    },
+                ]
+            },
+        )
+        @self.base_endpoint
+        async def get_collection_documents(
+            id: UUID = Path(
+                ..., description="The unique identifier of the collection"
+            ),
+            offset: int = Query(
+                0,
+                ge=0,
+                description="The number of documents to skip (for pagination)",
+            ),
+            limit: int = Query(
```

Comment:
We need to harmonize pagination across the API. It is extremely reasonable to want to pass a limit of -1 to get all entries.

I think it is helpful, too, to decide where we set the default. I have noticed inconsistencies in this across the codebase previosuly.